### PR TITLE
[ingress-nginx] Refactor startup hook

### PR DESCRIPTION
**Pull request by yalosev**
Signed-off-by: Yuriy Losev <yuriy.losev@flant.com>

## Description
Refactor startup hook, use kubeclient instead of snapshots

## Why do we need it, and what problem does it solve?
Snopshots don't work at startup phase. We have an extra run of the hook without any usefulness

## Changelog entries

```changes
section: ingress-nginx
type: fix
summary: Refactor startup hook
impact_level:  low
```

### CODE REVIEW

File: go_lib/dependency/dependency.go [hunk 1]
```diff
@@ -257,6 +263,27 @@ func (mdc *mockedDependencyContainer) SetK8sVersion(ver k8s.FakeClusterVersion)
 	mdc.K8sClient = cli
 }
 
+// CreateDynamicFromYAML create dynamic object from yaml string
+// you can create multiply objects of the same kind by separating yaml with '---'
+func (mdc *mockedDependencyContainer) CreateDynamicFromYAML(gvr schema.GroupVersionResource, data string) error {
+	arr := strings.Split(data, "---")
```
Comment (diafour) [inline go_lib/dependency/dependency.go:269]: This split pattern is error-prone. There is a 'ListFromYamlDocs' in [kube-client](https://github.com/flant/kube-client/blob/master/manifest/manifest_list.go#L5) that is used for monitoring Helm resources in addon-operator.

**Comment by diafour** [comment]
> Snopshots don't work at startup phase. We have an extra run of the hook without any usefulness

You can also use "Synchronization only OnKubernetes": remove onStartup and use OnKubernetes with `executeHookOnEvent: false` to get snapshots once at module startup.

File: go_lib/dependency/dependency.go (hunk 1 above)
Comment (yalosev) [reply go_lib/dependency/dependency.go]: Well. It was for tests only but you are right. I have fixed it

**Comment by yalosev** [comment]
seems like we don't need onStartup binding here at all

**Review by nabokihms — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

**Pull request merged**